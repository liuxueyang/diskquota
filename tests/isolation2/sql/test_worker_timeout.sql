!\retcode gpconfig -c diskquota.worker_timeout -v 1;
!\retcode gpstop -u;

SELECT gp_inject_fault_infinite('diskquota_worker_main', 'suspend', dbid)
  FROM gp_segment_configuration WHERE role='p' AND content=-1;

1&: SELECT diskquota.wait_for_worker_new_epoch();

SELECT pg_sleep(2 * current_setting('diskquota.worker_timeout')::int);

SELECT pg_cancel_backend(pid) FROM pg_stat_activity
WHERE query = 'SELECT diskquota.wait_for_worker_new_epoch();';

SELECT gp_inject_fault_infinite('diskquota_worker_main', 'resume', dbid)
  FROM gp_segment_configuration WHERE role='p' AND content=-1;

1<:

!\retcode gpconfig -r diskquota.worker_timeout;
!\retcode gpstop -u;
