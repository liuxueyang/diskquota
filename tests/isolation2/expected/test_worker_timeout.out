!\retcode gpconfig -c diskquota.worker_timeout -v 1;
(exited with code 0)
!\retcode gpstop -u;
(exited with code 0)

SELECT gp_inject_fault_infinite('diskquota_worker_main', 'suspend', dbid) FROM gp_segment_configuration WHERE role='p' AND content=-1;
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

1&: SELECT diskquota.wait_for_worker_new_epoch();  <waiting ...>

SELECT pg_sleep(2 * current_setting('diskquota.worker_timeout')::int);
 pg_sleep 
----------
          
(1 row)

SELECT pg_cancel_backend(pid) FROM pg_stat_activity WHERE query = 'SELECT diskquota.wait_for_worker_new_epoch();';
 pg_cancel_backend 
-------------------
 t                 
(1 row)

SELECT gp_inject_fault_infinite('diskquota_worker_main', 'resume', dbid) FROM gp_segment_configuration WHERE role='p' AND content=-1;
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

1<:  <... completed>
ERROR:  canceling statement due to user request

!\retcode gpconfig -r diskquota.worker_timeout;
(exited with code 0)
!\retcode gpstop -u;
(exited with code 0)
